Update C++ rules (#3220)
SonarSource/rspec: rules/S6168/cfamily/rule.adoc

== Why is this an issue?

``++std::jthread++``, introduced in {cpp}20, is a wrapper around ``++std::thread++``. This way, it has the same functionalities as ``++std::thread++``, making the substitution straightforward while adding two interesting behaviors:

* It joins by default in its destructor. If a ``++std::thread++`` was not joined or detached before being destroyed, a call to ``++std::terminate++`` was made. This behavior can’t happen with ``++std::jthread++``.

> * It can be canceled or stopped in some situations by calling ``++request_stop()++``

Reviewer: For consistency with the previous bullet point:
```suggestion
* It can be canceled or stopped in some situations by calling ``++request_stop()++``.
```

> This rule raises an issue as soon as ``++std::thread++`` is used.

Reviewer: I personally find "as soon as" quite heavy. Feel free to dismiss if you disagree, it's subjective.
```suggestion
This rule raises an issue when ``++std::thread++`` is used.
```

== Resources

* {cpp} reference - https://en.cppreference.com/w/cpp/thread/jthread[std::jthread]

> * {cpp} Core Guidelines - https://github.com/isocpp/CppCoreGuidelines/blob/036324/CppCoreGuidelines.md#cp25-prefer-gsljoining_thread-over-stdthread[CP.25 - Prefer "gsl::joining_thread" over "std::thread"]

Reviewer: Should these be in different categories? (Documentation and External coding guidelines) I personally don't mind.. feel free to dismiss.

Author: Indeed, that is what I usually do. Fixed in https://github.com/SonarSource/rspec/pull/3220/commits/7394fa6466cdf8fa779a7577e261a6c10cb38632
